src/lib/writers/LogWriter.ts

```diff
@@ -0,0 +1,29 @@
+import * as path from 'node:path';
+import * as fs from 'node:fs/promises';
+import {CodeAnalyzerConfig} from '@salesforce/code-analyzer-core';
+
+export interface LogWriter {
+	writeToLog(message: string): void;
+}
+
+export class LogFileWriter implements LogWriter {
+	private readonly writeStream: NodeJS.WritableStream;
+
+	private constructor(writeStream: NodeJS.WritableStream) {
+		this.writeStream = writeStream;
+	}
+
+	public writeToLog(message: string): void {
+		this.writeStream.write(message);
+	}
+
+	public static async fromConfig(config: CodeAnalyzerConfig): Promise<LogFileWriter> {
+		const logFolder = config.getLogFolder();
+		// Use the current timestamp to make sure each transaction has a unique logfile. If we want to reuse logfiles,
+		// or just have one running logfile, we can change this.
+		const logFile = path.join(logFolder, `sfca-${Date.now()}.log`);
+		// 'w' flag causes the file to be created if it doesn't already exist.
+		const fh = await fs.open(logFile, 'w');
+		return new LogFileWriter(fh.createWriteStream({}));
```

Comment:
Do we need to close the file handle eventually? Any consequences of not closing it? If you do need to close it, then maybe add a close method to the writer... and then add a stopListening method to your LogEventListener class.

Reply:
I believe the file handle automatically closes at the end of the process, so we don't strictly need to worry about closing it ourselves. However, according to the Internet, it's technically possible to have too many open file handles, and then you get an `EMFILE: Too many open files` error. I'm not sure what that limit is, but it seems implausible that we have to worry about reaching it.